haas/api.py

```diff
@@ -510,6 +510,23 @@ def headnode_detach_network(headnode, hnic):
                             # Network Code #
                             ################
 
+@rest_call('GET', '/networks')
+def list_networks():
+    """List all networks.
+    Returns a JSON array of strings representing a list of projects.
+    Example:  {"channels": ["101", "107"], "name": ["net-05", "qnet-00"]}
+    """
+    db = model.Session()
+    networks = db.query(model.Network).all()
+    result = {}
+    for n in networks:
+        if n.access is not None:
+            net = {'channel': n.network_id, 'project': n.access.label}
+        else:
+            net = {'channel': n.network_id, 'project': 'Public'}
```

Comment:
Nitpick, but if there's a project named "Public" this will be ambiguous.  Maybe do ``'public': True`` instead?